infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/SinkWriter.java

```diff
@@ -64,8 +70,12 @@ List<SMREntry> fetchNewEntries(List<SMREntry> smrEntries) {
             ByteBuf valueBuf = Unpooled.wrappedBuffer((byte[]) objs[1]);
             CorfuRecord<TableDescriptors, TableMetadata> record =
                     (CorfuRecord<TableDescriptors, TableMetadata>) protobufSerializer.deserialize(valueBuf, null);
+            UUID streamId = CorfuRuntime.getStreamID(TableRegistry.getFullyQualifiedTableName(tableName));
 
-            if (!registryTable.containsKey(tableName)) {
+            // There is potential data loss for Sink side streaming tasks if checking with registry table directly. If
+            // tables already opened but config has not been synced, new synchronization will not be triggered and those
+            // streams' tags could not be found from current config.
+            if (!config.getDataStreamToTagsMap().containsKey(streamId)) {
```

Comment:
why are we checking against the stream tags map and not streamsToReplicate?

Also, when new streams are found, they should be added to streamsToReplicate and the streaming config map.

Reply:
Resolved in an offline discussion